### ISSUE TITLE
[Bug]: The group_nodes of replica is empty after scale querynodes

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#23755]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20230426-5cd21893
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka): pulsar  
- SDK version(e.g. pymilvus v2.0.0rc2): 2.4.0.dev4
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. deploy milvus with 2 querynodes
2. create collection 2 shards -> insert -> create index
3. load collection with 2 replicas -> search successfully
4. scale querynode from 2 to 1
5. search and get_replicas failed as expected
6. scale querynode from 1 back to 2
7. release collectin -> load collection with 2 replicas
8. get_replicas return empty group_nodes
```
# Verify replica info is correct
            replicas = collection_w.get_replicas()[0]
            assert len(replicas.groups) == 2
            for group in replicas.groups:
>               assert len(group.group_nodes) == 1
E               assert 0 == 1
E                +  where 0 = len(())
E                +    where () = Group: <group_id:441061577189228547>, <group_nodes:()>, <shards:[Shard: <channel_name:scale-in-query-rootcoord-dml_1_441061575867236675v1>, <shard_leader:3>, <shard_nodes:[3]>, Shard: <channel_name:scale-in-query-rootcoord-dml_0_441061575867236675v0>, <shard_leader:3>, <shard_nodes:[3]>]>, <resource_group: __default_resource_group>, <num_outbound_node: {}>.group_nodes

test_query_node_scale.py:387: AssertionError
```

### Milvus Log

argo workflow name in devops cluster: scale-out-in-query-cron-1682515800
server pod after scale from 1 -> 2. in devops-cluster, chaos-testing namespace
```
scale-in-query-milvus-datacoord-5cc9845896-mh47x
scale-in-query-milvus-datanode-579ff67b9c-xxz7k
scale-in-query-milvus-indexcoord-b9df77df4-xx7nj
scale-in-query-milvus-indexnode-6886d7f9c-x9fqr
scale-in-query-milvus-proxy-6ccbb6b479-5hhgw
scale-in-query-milvus-querycoord-6d9d97b6b-7nj98
scale-in-query-milvus-querynode-7bffc65c5c-wbrcx
scale-in-query-milvus-querynode-7bffc65c5c-xxwxr
scale-in-query-milvus-rootcoord-55b48b88-tkts6
```

**Comment by xiaofan-luan** [comment]
this has to be fixed.
Probably just by a background checker

**Comment by weiliu1031** [comment]
first of all, the assign log logic is right,  logs can confirm that:
![image](https://user-images.githubusercontent.com/108661493/235045743-956960a8-1b16-4ecc-8f82-8df6a5274898.png)

then the verify fails due to #23396 , which issue two problems:
1. the node list in replica info hasn't been set, so you can't get the node num
2. the node info in shard replica has been set wrong value, it will saw the whole replica nodes list.

working on it

**Comment by weiliu1031** [comment]
please verify this @ThreadDao

**Comment by ThreadDao** [comment]
Fixed `master-20230518-7744573d`
